include/RMGGermaniumDetector.hh

```diff
@@ -48,6 +48,7 @@ class RMGGermaniumDetectorHit : public G4VHit {
     int detector_uid = -1;
     int particle_type = -1;
     float energy_deposition = -1;
+    float distance_to_surface = -1;
```

Comment:
```suggestion
    double distance_to_surface = -1;
```

> (It might be relevant to note that the original distance in Geant4 was saved in m)

it makes definitely sense that you see a difference on the 1e-7 level. you store it as `float` in m, and float has roughly 6-7 significant decimal digits.
To get the best resolution out (especially for the small distances here), storing as `double` increases the available precision.

this is also important, as we decide only later in the output scheme if we write out double or float. That makes no sense, if we do not have doubles in-memory (i.e. it would create "**false precision**" in the output file)

please also rebase on top of the current `main` branch, I added some compiler options to warn in those cases earlier today... (see #193)